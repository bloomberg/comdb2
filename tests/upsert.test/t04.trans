1 SELECT '1a' AS TEST
1 DROP TABLE IF EXISTS t
1 CREATE TABLE t(a int, b int)
1 CREATE UNIQUE INDEX ab ON t(a, b)
1 CREATE UNIQUE INDEX a ON t(a)
1 BEGIN
1 INSERT INTO t(a, b) VALUES(1, 1) ON CONFLICT(a) DO UPDATE SET b = 10
2 INSERT INTO t(a, b) VALUES(1, 2)
2 SELECT a, b FROM t
1 COMMIT
1 SELECT a, b FROM t

1 SELECT '1b' AS TEST
1 DROP TABLE IF EXISTS t
1 CREATE TABLE t(a int, b int)
1 CREATE UNIQUE INDEX a ON t(a)
1 CREATE UNIQUE INDEX ab ON t(a, b)
1 BEGIN
1 INSERT INTO t(a, b) VALUES(1, 1) ON CONFLICT(a) DO UPDATE SET b = 10
2 INSERT INTO t(a, b) VALUES(1, 2)
2 SELECT a, b FROM t
1 COMMIT
1 SELECT a, b FROM t

1 SELECT '2a' AS TEST
1 DROP TABLE IF EXISTS t
1 CREATE TABLE t(a int, b int)
1 CREATE UNIQUE INDEX ab ON t(a, b)
1 CREATE UNIQUE INDEX a ON t(a)
1 BEGIN
1 INSERT INTO t(a, b) VALUES(1, 1) ON CONFLICT(a) DO UPDATE SET b = 2
2 INSERT INTO t(a, b) VALUES(1, 1)
2 SELECT a, b FROM t
1 COMMIT
1 SELECT a, b FROM t

1 SELECT '2b' AS TEST
1 DROP TABLE IF EXISTS t
1 CREATE TABLE t(a int, b int)
1 CREATE UNIQUE INDEX a ON t(a)
1 CREATE UNIQUE INDEX ab ON t(a, b)
1 BEGIN
1 INSERT INTO t(a, b) VALUES(1, 1) ON CONFLICT(a) DO UPDATE SET b = 2
2 INSERT INTO t(a, b) VALUES(1, 1)
2 SELECT a, b FROM t
1 COMMIT
1 SELECT a, b FROM t
