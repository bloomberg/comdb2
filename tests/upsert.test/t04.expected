done
done
done
done
done
done
(rows inserted=1)
done
done
(a=1, b=10)
done
